### PR TITLE
feat: uffizzi integration

### PR DESCRIPTION
**Pull request by daramayis**
# Description
Uffizzi CI/CD integration.

### CODE REVIEW

File: docker-compose.uffizzi.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+services:
+  application:
+    image: "${APP_IMAGE}"
+    ports:
+      - "6006:6006"
+    restart: always
+    deploy:
+      resources:
+        limits:
+          memory: 4000m
```
Comment (axisofentropy) [inline docker-compose.uffizzi.yml]: I don't think the `m` notation makes sense here. Try the default `125M` and work upwards from there.  More info here https://docs.uffizzi.com/references/compose-spec/#resources-limits-memory

Also Uffizzi ignores `resources.cpu` specifications.

File: docker-compose.uffizzi.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  ingress:
+    service: application
+    port: 6006
+
+services:
+  application:
+    image: "${APP_IMAGE}"
+    ports:
+      - "6006:6006"
+    restart: always
```
Comment (axisofentropy) [inline docker-compose.uffizzi.yml]: I believe Uffizzi ignores this `restart` specification. But Kubernetes does always try to restart containers.

Comment (daramayis) [reply docker-compose.uffizzi.yml]: OK, I will remove it.

File: docker-compose.uffizzi.yml (hunk 1 above)
Comment (daramayis) [reply docker-compose.uffizzi.yml]: - `resources.cpu` has been removed.
- Tried following memory values
  - `125M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3318915889
  -  `500M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3319058470/jobs/5483736212
  - `1000M` - app preview deployment failed - https://github.com/daramayis/f-ui-kit/actions/runs/3319278317
  - `2000M` - app preview deployed successfully in 8 minutes.

File: .github/workflows/uffizzi-build.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
+name: Build PR Image
+on:
+  pull_request:
+    types: [opened,synchronize,reopened,closed]
+
+jobs:
+
+  build-appication:
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-build.yml]: typo

File: .github/workflows/uffizzi-build.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name != 'pull_request' || github.event.action != 'closed' }}
+    outputs:
+      tags: ${{ steps.meta.outputs.tags }}
+    steps:
+      - name: Checkout git repo
+        uses: actions/checkout@v3
+      - name: Generate UUID image name
+        id: uuid
+        run: echo "::set-output name=uuid::$(uuidgen)"
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-build.yml]: We're trying to deprecate `set-output`. the sooner the better.  Only change it here if this PR is likely to be accepted.  This is a good example https://github.com/UffizziCloud/quickstart/commit/3731ddb33771ddda88d93db2356dddf48f1ae7b6

We can also upgrade the third-party actions we're using like in this commit https://github.com/UffizziCloud/quickstart/commit/af17b37b9513f28e6299577611947ec8fe3cdade

File: .github/workflows/uffizzi-build.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: preview-spec
+          path: docker-compose.rendered.yml
+          retention-days: 2
+      - name: Serialize PR Event to File
+        run:  |
+          cat << EOF > event.json
+          ${{ toJSON(github.event) }} 
+
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-build.yml:67]: empty line could get into JSON file

File: .github/workflows/uffizzi-build.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+          retention-days: 2
+
+  delete-preview:
+    name: Call for Preview Deletion
+    runs-on: ubuntu-latest
+    if: ${{ github.event.action == 'closed' }}
+    steps:
+      # If this PR is closing, we will not render a compose file nor pass it to the next workflow.
+      - name: Serialize PR Event to File
+        run: echo '${{ toJSON(github.event) }}' > event.json
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-build.yml:83]: For quote safety this should be implemented as above on line `61`.

File: docker-compose.uffizzi.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+x-uffizzi:
+  ingress:
+    service: application
+    port: 6006
+
+services:
+  application:
+    image: "${APP_IMAGE}"
+    ports:
```
Comment (axisofentropy) [inline docker-compose.uffizzi.yml:11]: I don't think this does anything on Uffizzi, but it may be helpful for people reading this file. Optional.

**Comment by axisofentropy** [comment]
I added some comments but they're mostly optional changes. This PR looks good!

File: .github/workflows/uffizzi-preview.yml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        if: ${{ runner.debug }}
+        run: |
+          echo "PR number: ${{ steps.pr.outputs.number }}"
+          echo "Compose file hash: ${{ steps.hash.outputs.hash }}"
+          cat event.json
+  deploy-uffizzi-preview:
+    name: Use Remote Workflow to Preview on Uffizzi
+    needs:
+      - cache-compose-file
+    uses: UffizziCloud/preview-action/.github/workflows/reusable.yaml@v2
```
Comment (axisofentropy) [inline .github/workflows/uffizzi-preview.yml]: Fix to a specific version of this reusable workflow so we don't accidentally break this in the future. Right now that's `v2.6.0`.

File: .github/workflows/uffizzi-build.yml (hunk 3 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml]: Fixed.

File: .github/workflows/uffizzi-build.yml (hunk 5 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml]: Done.

File: .github/workflows/uffizzi-build.yml (hunk 4 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml]: Done.

File: .github/workflows/uffizzi-build.yml (hunk 6 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml]: Done.

File: .github/workflows/uffizzi-preview.yml (hunk 8 above)
Comment (daramayis) [reply .github/workflows/uffizzi-preview.yml]: Done

File: docker-compose.uffizzi.yml (hunk 7 above)
Comment (daramayis) [reply docker-compose.uffizzi.yml:11]: Yeah, it's helpful when we read this file :)

File: .github/workflows/uffizzi-build.yml (hunk 4 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml]: As we agreed - we will apply this change later.

File: .github/workflows/uffizzi-build.yml (hunk 5 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml:67]: As we agreed - we will apply this change later.

File: .github/workflows/uffizzi-build.yml (hunk 6 above)
Comment (daramayis) [reply .github/workflows/uffizzi-build.yml:83]: As we agreed - we will apply this change later.

File: .github/workflows/uffizzi-preview.yml (hunk 8 above)
Comment (daramayis) [reply .github/workflows/uffizzi-preview.yml]: Done.

**Pull request closed without merging**